scripts/buildPythonLibraries.sh

```diff
@@ -2,65 +2,86 @@
 
 # Utility script that will copy files over from your xml directory, then call
 # the salgenerator commands to build your SAL libraries. 
-XML_WORK_DIR="/gitdir/ts_xml" #Update to be the path to your xml repo.
+# optioanlly add  list or args to run after the Topic name e..g MTmount java lib 
+# validate and htmlo will be run any way -- defaul is python c++ lib as it was before this change.
+# Option -nc in FIRST position to not clean all the files from test
+ 
+if [ -z ${XML_WORK_DIR}x ]; then
+    XML_WORK_DIR="/gitdir/ts_xml" #Update to be the path to your xml repo.
+fi
 
+ALWAYS="validate html"
+CMDS="$ALWAYS python  cpp lib"  #adding sal taken care of below
 set -e
 if [ -z "$1" ]
   then
     echo "Must run with subsystem name (e.g. $ buildPythonTopics Test)"
     exit 1
 fi
+CLEAN=1
+if [ "$1" = "-nc" ]; then
+	CLEAN=0
+	shift
+fi
+T=$1
+
+if [ $# -gt 1 ] # Take remainng argumets as which generator phases to run after validate html 
+  then
+    shift
+    CMDS="$ALWAYS $*"
+fi
 
+echo "Running with CLEAN=$CLEAN TOPIC=$T and  $CMDS"
 printf "\nEntering build directory, copying over xml files\n"
 printf '%*s\n' "${COLUMNS:-$(tput cols)}" '' | tr ' ' -
 
 cd ${SAL_WORK_DIR}
 cp ${XML_WORK_DIR}/sal_interfaces/SALSubsystems.xml .
 cp ${XML_WORK_DIR}/sal_interfaces/SALGenerics.xml .
 
-if [ -s ${XML_WORK_DIR}/sal_interfaces/$1/$1_Telemetry.xml ]
-then
-	cp ${XML_WORK_DIR}/sal_interfaces/$1/$1_Telemetry.xml .
-fi
-
-if [ -s ${XML_WORK_DIR}/sal_interfaces/$1/$1_Commands.xml ]
-then
-	cp ${XML_WORK_DIR}/sal_interfaces/$1/$1_Commands.xml .
-fi
-
-if [ -s ${XML_WORK_DIR}/sal_interfaces/$1/$1_Events.xml ]
-then
-	cp ${XML_WORK_DIR}/sal_interfaces/$1/$1_Events.xml .
-fi
+for ft in "Telemetry" "Commands" "Events" ; do 
```

Comment:
CSC's are no longer required to have commands for salgenerator to run. You can run salgenerator without a commands.xml being file and it will create default ones.

Reply:
i may have another look

Reply:
you could have at least taken out the hard coded 
XML_WORK_DIR="/gitdir/ts_xml" #Update to be the path to your xml repo.